tests/backtest/test_trade_request.py

```diff
@@ -1,42 +1,159 @@
 """Test the TradeRequest class."""
 
+import pandas as pd
 import pytest
 
-from coin_test.backtest import TradeRequest
-from coin_test.util import AssetPair, Side, TradeType
+from coin_test.backtest import (
+    LimitTradeRequest,
+    MarketTradeRequest,
+    StopLimitTradeRequest,
+)
+from coin_test.util import AssetPair, Side
 
 
-def test_trade_request(example_asset_pair: AssetPair) -> None:
+def test_market_trade_request(asset_pair: AssetPair) -> None:
     """Initialize correctly."""
-    example_side = Side.BUY
-    example_trade_type = TradeType.MARKET
-    example_notional = 1000.0
+    side = Side.BUY
+    notional = 1000.0
 
-    x = TradeRequest(
-        example_asset_pair, example_side, example_trade_type, example_notional
-    )
+    x = MarketTradeRequest(asset_pair, side, notional)
 
-    assert x.asset_pair == example_asset_pair
-    assert x.side == example_side
-    assert x.type_ == example_trade_type
-    assert x.notional == example_notional
+    assert x.asset_pair == asset_pair
+    assert x.side == side
+    assert x.notional == notional
 
+    assert x.should_execute(999.99) is True
 
-def test_bad_trade_request(example_asset_pair: AssetPair) -> None:
+
+def test_market_trade_request_build_trade_buy_notional(
+    asset_pair: AssetPair, timestamp_asset_price: dict[AssetPair, pd.DataFrame]
+) -> None:
+    """Build Buy Trade with notional correctly."""
+    side = Side.BUY
+    notional = 1000.0
+
+    trade_request = MarketTradeRequest(asset_pair, side, notional)
+
+    trade = trade_request.build_trade(timestamp_asset_price)
+
+    trade_price = timestamp_asset_price[asset_pair]["High"].iloc[0]
+
+    assert trade.side == side
+    assert trade.asset_pair == asset_pair
+    assert trade.price == trade_price
+    assert trade.amount == notional // trade_price
+
+
+def test_market_trade_request_build_trade_buy_amount(
+    asset_pair: AssetPair, timestamp_asset_price: dict[AssetPair, pd.DataFrame]
```

Comment:
Does timestamp_asset_price include a timestamp? If not, maybe rename to asset_price